codec/src/main/java/io/netty5/handler/codec/DelimiterBasedFrameDecoder.java

```diff
@@ -329,15 +329,15 @@ private static int indexOf(ByteBuf haystack, ByteBuf needle) {
 
             if (needleIndex == needle.capacity()) {
                 // Found the needle from the haystack!
-                return i - haystack.readerIndex();
+                return i - haystack.readerOffset();
             }
         }
         return -1;
     }
 
-    private static void validateDelimiter(ByteBuf delimiter) {
+    private static void validateDelimiter(Buffer delimiter) {
         requireNonNull(delimiter, "delimiter");
-        if (!delimiter.isReadable()) {
+        if (delimiter.readableBytes() <= 0) {
```

Comment:
This cant be negative...
```suggestion
        if (delimiter.readableBytes() == 0) {
```

Reply:
:woman_facepalming:

Reply:
thanks